curator-client/src/main/java/org/apache/curator/ConnectionState.java

```diff
@@ -218,7 +216,7 @@ public String call()
         };
         int lastNegotiatedSessionTimeoutMs = getLastNegotiatedSessionTimeoutMs();
         int useSessionTimeoutMs = (lastNegotiatedSessionTimeoutMs > 0) ? lastNegotiatedSessionTimeoutMs : sessionTimeoutMs;
-        ConnectionHandlingPolicy.CheckTimeoutsResult result = connectionHandlingPolicy.checkTimeouts(hasNewConnectionString, connectionStartMs, useSessionTimeoutMs, connectionTimeoutMs);
+        CheckTimeoutsResult result = checkTimeouts(hasNewConnectionString, connectionStartMs, useSessionTimeoutMs, connectionTimeoutMs);
```

Comment:
To my eye, the separate method doesn't add much and hurts readability. To the point, `RESET_CONNECTION`, `CONNECTION_TIMEOUT` and `SESSION_TIMEOUT` are no longer possible. Only `NOP` and `NEW_CONNECTION_STRING` are possible so `CheckTimeoutsResult` is no longer needed.

Reply:
Yes. I mentioned it in the JIRA and commit message.

>Besides, some values in CheckTimeoutsResult seems out of day that we might have a follow-up to handle it.

I'd prefer do it in another cycle so that I have cycle to understand the impact by removing it :-)

Reply:
If you think we'd better include the removal of `CheckTimeoutsResult` in this pr, we can hold it and let me try to investigate in time/

Reply:
Yes, it would be better in this PR